tests/ctl/core/test_api.py

```diff
@@ -142,6 +144,41 @@ def test_api_upsert(
         )
         assert result.status_code == 200
 
+    async def test_upsert_validates_resources_against_pydantic_model(
+        self, test_config: FidesConfig, resources_dict: Dict, async_session
+    ):
+        endpoint = "dataset"
+        manifest: Dataset = resources_dict[endpoint]
+        dict_manifest = manifest.dict()
+        del dict_manifest["organization_fides_key"]
+
+        result = _api.upsert(
+            url=test_config.cli.server_url,
+            headers=test_config.user.request_headers,
+            resource_type=endpoint,
+            resources=[dict_manifest],
+        )
+        assert result.status_code == 200
+
+        resource = await get_resource(Dataset, manifest.fides_key, async_session)
+        assert resource.organization_fides_key == "default_organization"
+
+    async def test_data_categories_validated_against_db(
+        self, test_config: FidesConfig, resources_dict: Dict, async_session
+    ):
+        endpoint = "dataset"
+        manifest: Dataset = resources_dict[endpoint]
+        dict_manifest = manifest.dict()
+        dict_manifest["collections"][0]["data_categories"] = ["bad_category"]
+
+        result = _api.upsert(
```

Comment:
do you think it's worth testing the create and update too?

Reply:
sure can, thanks Allison!